packages/cloudflare/src/api/get-cloudflare-context.ts

```diff
@@ -0,0 +1,50 @@
+import "server-only";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface CloudflareEnv {}
+}
+
+export type CloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+> = {
+  /**
+   * object containing the worker's bindings ([docs](https://developers.cloudflare.com/workers/runtime-apis/bindings/))
+   */
+  env: CloudflareEnv;
+  /**
+   * object containing information regarding the current request ([docs](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties))
+   */
+  cf: CfProperties;
+  /**
+   * the current execution context object ([docs](https://developers.cloudflare.com/workers/runtime-apis/context))
+   */
+  ctx: Context;
+};
+
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
+ *
+ * @returns the cloudflare context
+ */
+export async function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(): Promise<CloudflareContext<CfProperties, Context>> {
+  const cloudflareContext = (
```

Comment:
nit: a more concise version could be `const cloudflareContext = (globalThis as any)[cloudflareContextSymbol] as CloudflareContext<CfProperties, Context> | undefined;`

Reply:
ok thanks, if you're ok with it let's keep it as is and we can always revisit it later on 🙂